src/views/PopularRepositories.vue

```diff
@@ -1,20 +1,64 @@
 <template>
   <div>
     <h3 class="center-align">Popular Repositories</h3>
-    <slot />
-    <Repositories />
+    <p class="center">
+      Showing <strong>{{ repositories.length.toLocaleString() }}</strong> repositories
+      <span>sorted by stars.</span>
+    </p>
+    <InputSearch label="Filter repository by name or description..." v-model="searchTerm" />
+    <Repositories v-for="repository in filteredRepositories" :repository="repository" :key="repository.id" />
   </div>
 </template>
 
 <script>
+import { mapGetters } from "vuex";
+import * as JsSearch from "js-search";
+import sortBy from "lodash/sortBy";
 import Repositories from "@/components/RepositoryCard";
+import InputSearch from "@/components/InputSearch";
+const searcher = new JsSearch.Search("position");
```

Comment:
Would we be able to share the searcher between more components? If so, wouldn't we want to extract this into it's own module and have consumers define the indexes and documents?

Like, a searcher module that returns a function that returns a new instance of the searched and sets up the search index and the strategy by default.

Reply:
I agree, so I created a factory for this.

Fixed: https://github.com/VueDominicana/open-source/pull/36/commits/0fe11fab0cd6c75370189ace353d67f101a6ef88 and https://github.com/VueDominicana/open-source/pull/36/commits/056dc73ef30335c70b25740fe1ff8fe253953290